transports/alltransports/alltransports.go

```diff
@@ -0,0 +1,29 @@
+package alltransports
+
+import (
+	"strings"
+
+	// register all known transports
+	_ "github.com/containers/image/directory"
+	_ "github.com/containers/image/docker"
+	_ "github.com/containers/image/docker/daemon"
+	_ "github.com/containers/image/oci/layout"
+	_ "github.com/containers/image/openshift"
+	_ "github.com/containers/image/storage"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+	"github.com/pkg/errors"
+)
+
+// ParseImageName converts a URL-like image name to a types.ImageReference.
+func ParseImageName(imgName string) (types.ImageReference, error) {
```

Comment:
The underlying idea is that the strings are exposed to the users, so they should _always_ mean the same thing; we should not have `skopeo` and `whateverothertool` use strings which _look_ the same but actually each accepting a different subset of transports. So, if any user of the library uses the strings in the UI and wants to use `ParseImageName`, they automatically get all transports: they can’t forget the `_` import, as you say, and they can’t even opt out — this automatically tries to do the right thing for users. I understand that this may feel annoyingly heavy-handed; is it actually problematic? I _think_ that on balance this does the right thing for users but I can of course be mistaken.

(More specialized users of `containers/image` can use `somepackage.Transport.ParseReference` to get a part of the parsing functionality, or `somepackage.NewReference` if working with the native values directly.)